Add startosis version of the module

gbouv opened the pull request:
The only thing that cannot be achieved right now is to pass in the number of services to start as params of the CLI command. But this is something that can easily be tweaked by updating the `PARAM` section of the script before running it

h4ck3rk3y commented on kurtosis-module/impl/datastore_army_kurtosis_module.go:
```diff
@@ -44,13 +44,14 @@ func (module *DatastoreArmyKurtosisModule) Execute(enclaveCtx *enclaves.EnclaveC
 
 // ====================================================================================================
-//                                       Private helper functions
+//
```
gofmt screw up

h4ck3rk3y commented on startosis/datastore-army-module.star:
```diff
@@ -0,0 +1,43 @@
+    add_service(service_id = unique_service_id, service_config = service_config)
+
+def add_multiple_datastore_services(num_datastores):
+    for i in range(num_datastores):
```
nit: s/i/index/g

h4ck3rk3y reviewed (approved):
This looks alright but how does this get executed at the moment, you clone this and run the cli on it? Or do we run an interpreter that loads this? Or will the go sdk call this module?

gbouv commented:
Once the [CLI PR is merged](https://github.com/kurtosis-tech/kurtosis/pull/166), you will be able to run it with a simple:
```
kurtosis startosis exec datastore-army-module.star [--enclave-is datastore-army-module]
```

h4ck3rk3y reviewed (changes requested):
needs a CHANGELOG entry, preferably breaking

h4ck3rk3y commented:
> Once the [CLI PR is merged](https://github.com/kurtosis-tech/kurtosis/pull/166), you will be able to run it with a simple:
> 
> ```
> kurtosis startosis exec datastore-army-module.star [--enclave-is datastore-army-module]
> ```

Oh so you clone it and then do it. Just UX doesn't have to be a part of this at all but what about something like kurtosis startosis exec github.com/path/to/main.star, that loads and executes the main function?

mieubrisse commented on startosis/datastore-army-module.star:
```diff
@@ -0,0 +1,43 @@
+    service_config = struct(
+        container_image_name = datastore_image,
+        used_ports = {
+            datastore_port_id: struct(number = datastore_port_number, protocol = "TCP")
```
Magic `TCP` constant

mieubrisse replied:
ideally this is an enum that Starlark easily provides (don't have to do for now though)

gbouv replied:
Yes we can interpret Starlark scripts "in an environment", in which we can put all sort of stuffs. We should start thinking of what a good "Startosis base environment" looks like. Alternatively, we can use modules.

mieubrisse commented on startosis/datastore-army-module.star:
```diff
@@ -0,0 +1,43 @@
+
+module_name_for_logging = "datastore_army_module"
+
+datastore_image = "kurtosistech/example-datastore-server"
```
Recommend using UPPER_SNAKE_CASE convention for constants (will be a good habit to get into as we get bigger scripts)

gbouv commented:
@h4ck3rk3y yes we could add the execution of a script directly from GH, that'd be a FR :P
Added changelog, though I think Breaking is a bit too much here, we're not breaking anything of the current behaviour. I added a Feature for now. Once/When we remove the Go module code, we can do Breaking

mieubrisse reviewed (approved):
Good for me

h4ck3rk3y reviewed (approved):
Yalla @gbouv
